armory/datasets/config_load.py

```diff
@@ -0,0 +1,85 @@
+"""
+Temporary file for testing loading from config without modifying armory.utils
+"""
+
+import copy
+
+from armory.datasets import load, preprocessing, generator
+
+
+def actual_loader(
+    name,
+    version=None,
+    split="test",
+    shuffle_files=False,
+    framework="numpy",
+    epochs=1,
+    drop_remainder=False,
+    num_batches=None,
+    batch_size=1,
+    shuffle_elements=False,
+):
+    info, ds_dict = load.load(name, version=version, shuffle_files=shuffle_files)
+    preprocessor = preprocessing.get(name, version=version)
+    return generator.ArmoryDataGenerator(
+        info,
+        ds_dict,
+        split=split,
+        batch_size=batch_size,
+        framework=framework,
+        epochs=epochs,
+        element_filter=None,
+        element_map=preprocessor,
+        shuffle_elements=shuffle_elements,
+    )
+
+
+def load_dataset(dataset_config, *args, check_run=False, **kwargs):
+    """
+    Designed to be a drop-in replacement for armory.utils.config_loading.load_dataset
+
+    NOTE: very ugly
+    """
+    dataset_config = copy.deepcopy(dataset_config)
+    module = dataset_config.pop("module")
+    if module == "armory.data.datasets":
+        pass
+    elif module == "armory.data.adversarial_datasets":
+        pass  # TODO: check
+    else:
+        raise NotImplementedError
+
+    name = dataset_config.pop("name")
+    if ":" in name:
+        name, version = name.split(":")
+    else:
+        version = None
+
+    for k in "eval_split", "train_split":
+        kwargs.pop(k, None)
+
+    kwargs.update(dataset_config)
```

Comment:
the 
```
kwargs.update(dataset_config)
```

I believe should occur before the `for k in "eval_split", "train_split":`, since the former can add keys such as `'eval_split"` to `kwargs` which is then subsequently erroneously passed to `actual_loader()`

Reply:
Made the whole thing a bit more readable.